### PR TITLE
JBPM-8977: BPMN Editor removes output variables from task without alerting or throwing error

### PR DESCRIPTION
**Pull request by handreyrc**
@romartin, @inodeman Could you please review this PR?

### CODE REVIEW

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
I'm not really sure yet if should support duplicate names in the data outputs.. you're assuming not, but let's double check it plz (see [comment](https://issues.redhat.com/browse/RHPAM-2521?focusedCommentId=13948634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13948634))

**Comment by domhanak** [comment]
@romartin I gave it a thought and since we do not support duplicate DI we shouldn't support duplicate DO's. 
I really do not see a use case where user would like to have two identically named DO bound to some process variables. I am for showing an exception as we do with duplicate DI.

**Comment by romartin** [comment]
Hey @handreyrc @domhanak 
Well from [this comment](https://issues.redhat.com/browse/RHPAM-2521?focusedCommentId=13948678&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13948678) from @krisv , it should be better supporting duplicates on DO. Plz can you update the PR @handreyrc and let us know when it's ready again for testing? 
Thx!

**Comment by domhanak** [comment]
@romartin Could you please clarify the comment from Kris? 
From customer perspective, I would shy away from overwriting anything in case of multiple DP's with same name. Futhermore, I really can't see a use case with two DO's with equal names. Could you please give an example?

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by krisv** [comment]
@domhanak Imho it could make sense to map the same output parameter to different variables.  See for example:
https://github.com/kiegroup/kie-wb-playground/blob/7.0.x/itorders/src/main/resources/org/jbpm/demo/itorders/orderhardware.bpmn2#L231
There we mapped the manager approval boolean param to (1) a process variable approval and (2) a case variable manager decision.

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
@romartin @domhanak @krisv:  I updated the PR in order to support duplicates on DO as agreed. 

@romartin @inodeman: Could you review this PR please?

@romartin @hasys: SonarCloud is complaining about duplicated code. However, it isn't a real issue in my understanding. It is happening due to the code I replicated on the Kogito side. Is there any recommend proceeding to overcome the problem?

**Comment by romartin** [comment]
Hey @handreyrc 
About sonar cloud failures, agree, we assume we still have some code duplicatoin, so we cannot fix those failures now, no worries, good job anyway, thanks!  :+1:

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/ParsedAssignmentsInfo.java [hunk 1]
```diff
@@ -196,15 +197,30 @@ public AssociationList getAssociations() {
...
                 .getDeclarations()
-                .stream()
-                .map(varDecl -> InitializedVariable.outputOf(
-                        parentId,
-                        variableScope,
-                        varDecl,
-                        associations.lookupOutput(varDecl.getTypedIdentifier().getName())))
-                .collect(Collectors.toList());
+                .forEach(varDecl -> {
+                    if (associations.lookupOutputs(varDecl.getTypedIdentifier().getName()).isEmpty()) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/ParsedAssignmentsInfo.java]: All code inside the `forEach` is exaclly the same for backend as for client side marshallers, right? If so, would be possible to move it to some util class, in the `bpmn-api` module, so will be available on both sides, and we'll be able to write it just once. Makes sense?

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/ParsedAssignmentsInfo.java (hunk 1 above)
Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/ParsedAssignmentsInfo.java]: @romartin I worked on moving the code to a bmpn-api. However,  in this specific case there are a few classes consumed by the method that also need to be moved to get the method working. I could't get it working without creating more duplicated code. I believe the code is good to go as is. Any way I'm opened to new ideas.

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Pull request merged**